doc/Programming-guide/Your-first-Kaa-application/attach/demo-cpp/KaaDemo.cpp

```diff
@@ -56,6 +57,11 @@ class ConfigurationCollection : public kaa::IConfigurationReceiver {
     {
         // Run the Kaa endpoint.
         kaaClient_->start();
+        // Read default sample period
+        samplePeriod_ = kaaClient_->getConfiguration().samplePeriod;
```

Comment:
I tried it before, it doesn't work.
It must be after kaaClient_->start();
Because the function kaaClient_->getConfiguration() calls this function:
void KaaClient::checkClientState(State expected, const std::string& message)
{
    if (clientState_ != expected) {
        throw KaaException(message);
    }
}
Note! Expected is State::STARTED and clientState_ is other